pkg/ui/text.go

```diff
@@ -61,11 +61,15 @@ func (t Text) Index(k interface{}) (interface{}, error) {
 	}
 }
 
-// Concat implements Text+string, Text+Segment and Text+Text.
+// Concat implements Text+string, Text+Segment and Text+Text. Where "string"
+// refers to strings and data types, such as float64, where it is reasonable
+// to implicitly convert the type to a string.
```

Comment:
I'd rather not twist the definition of "string" here, and just document that this implements Text+string and Text+float64.

If the need arises for more types, it might be more convenient to say something like "string-like value" or "value convertible to string", but now it's just two types.